pkg/ir/ir.go

```diff
@@ -169,3 +170,23 @@ func (sql *NormalStmt) IsExtended() bool { return false }
 
 // GetOriginalSQL returns the original SQL statement used to get current IR result
 func (sql *NormalStmt) GetOriginalSQL() string { return string(*sql) }
+
+// ShowTrainStmt get and output the original train sql for ModelName
+type ShowTrainStmt struct {
+	// OriginalSQL is the SHOW TRAIN stmt itself
+	OriginalSQL string
+	// The model to show the train sql
+	ModelName string
+}
+
+// Execute generates and executes code for ShowTrainStmt
+func (sql *ShowTrainStmt) Execute(s Executor) error { return s.ExecuteShowTrain(sql) }
```

Comment:
I vaguely remember that I had a discussion with @typhoonzero a few months ago about do we need the Go interface in the design of IR. I don't member the conclusion and I think I might have preferred not to use interface.  Do you @typhoonzero remember the conclusion?

Reply:
Is the discussion details in https://github.com/sql-machine-learning/sqlflow/issues/1434? The current `Executor` interface still needed because we  have different execution types:

1. local: run codegen and execute the generated Python code locally.
1. pai: generate code and tar the code into a tarball then call PAI command to submit a job
1. alias: generate code and tar the code into a tarball and upload this tarball to OSS then call alias HTTP API call to submit a job.

So we should have one interface to derive these submitting operations and we should move these code out of `ir.go`. In current implementation, should move `ShowTrainStmt` 's member functions to `submitter.go`.

By the way, the current naming is quite confusing, e.g. "submitter" should be a Python API, it should not appear in Go. `Executor` is also confusing, we should go on work with https://github.com/sql-machine-learning/sqlflow/issues/1434